File: aesdukpt/src/main/java/com/joyner/aesdukpt/dukpt/ValuesAesDukpt.java
package com.joyner.aesdukpt.dukpt;

import com.security.aesdukpt.KeyType;

public class ValuesAesDukpt {
    private String[] gIntermediateDerivationKeyRegister;
    private boolean[] gIntermediateDerivationKeyInUse;
    private int gCurrentKey;
    private byte[] gDeviceID;
    private long gCounter;
    private long gShiftRegister;
    private KeyType gDeriveKeyType;

    public String[] getgIntermediateDerivationKeyRegister() {
        return gIntermediateDerivationKeyRegister;
    }

    public void setgIntermediateDerivationKeyRegister(String[] gIntermediateDerivationKeyRegister) {
        this.gIntermediateDerivationKeyRegister = gIntermediateDerivationKeyRegister;
    }

    public boolean[] getgIntermediateDerivationKeyInUse() {
        return gIntermediateDerivationKeyInUse;
    }

    public void setgIntermediateDerivationKeyInUse(boolean[] gIntermediateDerivationKeyInUse) {
        this.gIntermediateDerivationKeyInUse = gIntermediateDerivationKeyInUse;
    }

    public int getgCurrentKey() {
        return gCurrentKey;
    }

    public void setgCurrentKey(int gCurrentKey) {
        this.gCurrentKey = gCurrentKey;
    }

    public byte[] getgDeviceID() {
        return gDeviceID;
    }

    public void setgDeviceID(byte[] gDeviceID) {
        this.gDeviceID = gDeviceID;
    }

    public long getgCounter() {
        return gCounter;
    }

    public void setgCounter(long gCounter) {
        this.gCounter = gCounter;
    }

    public long getgShiftRegister() {
        return gShiftRegister;
    }

    public void setgShiftRegister(long gShiftRegister) {
        this.gShiftRegister = gShiftRegister;
    }

    public KeyType getgDeriveKeyType() {
        return gDeriveKeyType;
    }

    public void setgDeriveKeyType(KeyType gDeriveKeyType) {
        this.gDeriveKeyType = gDeriveKeyType;
    }
}

File: aesdukpt/src/main/java/com/joyner/aesdukpt/dukpt/ValuesAesDukpt.kt
package com.joyner.aesdukpt.dukpt

import com.security.aesdukpt.KeyType

class ValuesAesDukpt {
    private lateinit var gIntermediateDerivationKeyRegister: Array<String>
    private lateinit var gIntermediateDerivationKeyInUse: BooleanArray
    private var gCurrentKey = 0
    private lateinit var gDeviceID: ByteArray
    private var gCounter: Long = 0
    private var gShiftRegister: Long = 0
    private var gDeriveKeyType: KeyType? = null
    fun getgIntermediateDerivationKeyRegister(): Array<String> {
        return gIntermediateDerivationKeyRegister
    }

    fun setgIntermediateDerivationKeyRegister(gIntermediateDerivationKeyRegister: Array<String>) {
        this.gIntermediateDerivationKeyRegister = gIntermediateDerivationKeyRegister
    }

    fun getgIntermediateDerivationKeyInUse(): BooleanArray {
        return gIntermediateDerivationKeyInUse
    }

    fun setgIntermediateDerivationKeyInUse(gIntermediateDerivationKeyInUse: BooleanArray) {
        this.gIntermediateDerivationKeyInUse = gIntermediateDerivationKeyInUse
    }

    fun getgCurrentKey(): Int {
        return gCurrentKey
    }

    fun setgCurrentKey(gCurrentKey: Int) {
        this.gCurrentKey = gCurrentKey
    }

    fun getgDeviceID(): ByteArray {
        return gDeviceID
    }

    fun setgDeviceID(gDeviceID: ByteArray) {
        this.gDeviceID = gDeviceID
    }

    fun getgCounter(): Long {
        return gCounter
    }

    fun setgCounter(gCounter: Long) {
        this.gCounter = gCounter
    }

    fun getgShiftRegister(): Long {
        return gShiftRegister
    }

    fun setgShiftRegister(gShiftRegister: Long) {
        this.gShiftRegister = gShiftRegister
    }

    fun getgDeriveKeyType(): KeyType? {
        return gDeriveKeyType
    }

    fun setgDeriveKeyType(gDeriveKeyType: KeyType?) {
        this.gDeriveKeyType = gDeriveKeyType
    }
}